Respect-`show_avatars`-comments-#2248

BE-Webdesign opened the pull request:
Possible fix for #2248.  Respects the show_avatars option when
retrieving avatars data for comments. Modeled after #2151.

BE-Webdesign committed:
Respect-`show_avatars`-comments-#2248

Possible fix for #2248.  Respects the show_avatars option when
retrieving avatars data for comments. Modeled after #2151.

BE-Webdesign committed:
Attempt to solve test failures

Attempting to solve test failures.

rmccue commented on lib/endpoints/class-wp-rest-comments-controller.php:
```diff
@@ -899,6 +885,29 @@ public function get_item_schema() {
 			),
 		);
+
+		if ( get_option( 'show_avatars' ) ) {
```
This would be better off as a strict check (`=== true`) if we can.

BE-Webdesign replied:
Okay, will fix.  In the codex it is stored as 1 and 0 integer type.  Is this changed to a boolean or should I check for 1 === get_option( 'show_avatars' )?

rmccue commented:
Looks pretty good :+1: Small change in the check, but otherwise seems pretty sensible.

BE-Webdesign committed:
Second attempt to fix tests

Added strict check for get_option( 'show_avatars' ) should it be a
boolean?  Attempt to fix testing.

BE-Webdesign committed:
Third Attempt

Third attempt.

BE-Webdesign committed:
Fourth attempt

Fourth attempt

BE-Webdesign commented:
Hallelujah!

joehoyle commented on lib/endpoints/class-wp-rest-comments-controller.php:
```diff
@@ -899,6 +885,29 @@ public function get_item_schema() {
 			),
 		);
+
+		if ( '1' === get_option( 'show_avatars' ) ) {
```
I think this should be a `if ( get_option...` the strict type check, while correct for most cases, is somewhat unintuitive, as this _really_ should be a bool. +1 for allowing any truthy value.

joehoyle commented:
Should we also apply the same to the users controller? The database option name is `show_avatars` which makes it seem like it's not comment specific, though I understand that's where the option is in the dashbaord?

BE-Webdesign commented:
@joehoyle I believe this was already done in users #2151.  @rmccue suggested the strict check so if == is the way over === we can decide that in the meeting maybe?

danielbachhuber commented:
I'm :-1: on a strict check in this context. A filter from a plugin or theme should be able to return a truthy or falsy value, and see the API work as expected.

BE-Webdesign commented:
@danielbachhuber So do you believe we should put it back to:

if ( get_option( 'show_avatars' ) )
**or use**
if (  true == get_option( 'show_avatars' ) )
**or something else entirely?**

danielbachhuber commented:
I think we should use `if ( get_option( 'show_avatars' ) )`

rmccue commented:
I'm OK with the truthy/loose check so long as it doesn't cause side effects (i.e. we don't accidentally catch `'n'` or something).

danielbachhuber commented:
> I'm OK with the truthy/loose check so long as it doesn't cause side effects

Core uses `if ( ! get_option( 'show_avatars' ) )`: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/pluggable.php#L2356

I think we're fine using `if ( get_option( 'show_avatars' ) )`. @BE-Webdesign if you want to make that final change, we can get this landed.

BE-Webdesign commented:
Okay, I will change it back give me a second.

BE-Webdesign committed:
Merge remote-tracking branch 'refs/remotes/WP-API/develop' into Respect-`show_avatars`-comments#2248

BE-Webdesign committed:
Removing strict check.

Removing strict check get_option( 'show_avatars' ).

danielbachhuber commented:
Build fails because of https://github.com/WP-API/WP-API/issues/1488#issuecomment-184664633

Merging because curies need to be fixed in this release regardless.
